src/amber/router/params.cr

```diff
@@ -1,20 +1,14 @@
 module Amber::Router
-  class Params < Hash(String, String)
-    alias KeyType = String | Symbol
-    property files = {} of String => Amber::Router::Files::File
+  struct Params
+    property files = {} of String => File
+    property store : HTTP::Params = HTTP::Params.new(Hash(String, Array(String)).new)
 
-    def json(key : KeyType)
-      JSON.parse(self[key]?.to_s)
+    forward_missing_to @store
+
+    def json(key)
+      JSON.parse(store[key]?.to_s)
```

Comment:
I have a few questions about this.

- This seems like it's going to run once for each `parms.json`. Can it make a memo somewhere and only do that once?
- Is the idea here that a part of the params would be json but the rest wouldn't? I'm not sure I understand.
- Can this failure case be handled more gracefully? I think there should be a more gentle way to signal to a framework user that the json payload is invalid than raising an anonymous exception. Is that something that can be added?

Reply:
@eliasjpr and I have already talked about changes I'm going to make to this class to bring it back in line with what is currently in Amber. We can address this there too.

Reply:
Does that mean that this will be a breaking change now, and also later when your changes merge in? Why not just address the issue and not have any broken state?

Reply:
It's in master so it won't be released as a breaking change. As it stands I guess it is a tiny bit. For instance before this params[] could take a symbol or string as key. This removes the functionality until I add it back.

Reply:
> This seems like it's going to run once for each parms.json. Can it make a memo somewhere and only do that once?

This is to handle the case where json params go multiple levels deep. That can't be store in the normal params as JSON::TYPE so we store the nested values as a string that is parsed if you need it.

It would be possible to create `@json_params : Hash(String, JSON::Type)` which could then be set when `.json `is called.

```cr
def json(key)
  @json_params[key] ||= JSON.parse(store[key]?)
end
```